### ISSUE TITLE
Improve GUI status prompting mechanism

### ISSUE THREAD

**Issue by davidlatwe** [getavalon/core#435]
### Problem

I found that Avalon's GUI tools use a common widget function called `echo` to prompt status messages, [some of them](https://github.com/getavalon/core/blob/1d898998a6906fa46493cc8d83dbd25601471b77/avalon/tools/loader/app.py#L252) were able to show on widget, [some of them](https://github.com/getavalon/core/blob/master/avalon/tools/loader/widgets.py#L280) were print out only. I think this could be improved.

### Propose

Use `logging` to pipe status logs into a custom status line widget.

![statusline](https://user-images.githubusercontent.com/3357009/63571740-ae8ffa80-c5b3-11e9-8016-378ebbc316c4.gif)

By connecting Logger, we could prompt log message from anywhere to not only console but also GUI.

[Full code for demo](https://gist.github.com/davidlatwe/6188eb45a11b94ad2788caa0c8626d13)

NOTE: The implementation above, will not show messages in status line if log level is lower then previous message that was higher then `logging.WARNING`.

**Comment by BigRoy** [comment]
Could this also allow to expand and see the full log history in something like Maya's script editor? So we create our own window to show that.

**Comment by davidlatwe** [comment]
Yeah, if we also pipe logs into a log file, or just keeping them in memory, could do !

**Comment by mottosso** [comment]
Very nice. Could be a good starting point for shared widgets.

- `avalon\tools\widgets.py`

**Comment by davidlatwe** [comment]
> Could this also allow to expand and see the full log history in something like Maya's script editor?

About full log history, which one would be better for this ? A tree view widget like the one in pyblish-qml or lite that you could filter out logs by level and even sorting them, or just plain text like Maya script editor ?
